src/reducers/user_reducer.js

```diff
@@ -0,0 +1,34 @@
+import types from '../action/type'
+
+const DEFAULTSTATE={
+   user:null,
+   cart:null,
+   checkout:null,
+   loggedIn: false
+}
+export default (state = DEFAULTSTATE,action)=>{
+  switch(action.type){
+    case types.ACTIVE_USER_LOGIN:
+      return{
+        user:action.payload,
```

Comment:
You should always be adding the current state to the new state by using the spread operator. This will erase parts of your user state in its current form.

EXAMPLE:
```JAVASCRIPT
{
    ...state,
    user: action.payload,
    loggedIn: true
}
```